File: src/main/java/com/hoi4utils/clausewitz/code/effect/EffectSchema.java
package com.hoi4utils.clausewitz.code.effect;

import com.hoi4utils.clausewitz.code.scope.ScopeType;
import scala.collection.mutable.ListBuffer;

import java.util.EnumSet;
import java.util.List;

public record EffectSchema(List<String> pdxIdentifiers, EnumSet<ScopeType> supportedScopes,
                           EnumSet<ScopeType> supportedTargets) {

    public EffectSchema(String pdxIdentifier, EnumSet<ScopeType> supportedScopes) {
        this(scala.collection.immutable.List(pdxIdentifier), supportedScopes, null);
    }
}

File: src/main/scala/com/hoi4utils/clausewitz/code/effect/EffectSchema.scala
package com.hoi4utils.clausewitz.code.effect

import com.hoi4utils.clausewitz.code.scope.ScopeType

import scala.collection.mutable.ListBuffer


final class EffectSchema(pdxIdentifiers: List[String], supportedScopes: Set[ScopeType], supportedTargets: Set[ScopeType]) {
  def this(pdxIdentifier: String, supportedScopes: Set[ScopeType]) = {
    this(List(pdxIdentifier), supportedScopes, Set.empty)
  }

  def this(pdxIdentifier: String, supportedScopes: Option[Set[ScopeType]], supportedTargets: Option[Set[ScopeType]]) = {
    this(List(pdxIdentifier), supportedScopes.get, supportedTargets.get)
  }
}
